cdap-ui/app/cdap/components/PipelineDetails/store/index.js

```diff
@@ -109,9 +111,15 @@ const pipelineDetails = (state = DEFAULT_PIPELINE_DETAILS, action = defaultActio
           version: pipeline.artifact.version,
           scope: pipeline.artifact.scope
         },
-        config: {...newPipelineConfig}
+        config: {...newPipelineConfig},
```

Comment:
This might be a problem and the store might end up being unusable if the `JSON.parse` throws an exception. We should handle the JSON.parse failure inside the catch and return a state that corresponds to that

Reply:
Have added a line to set `newPipelineConfig` back to `{...state.config}` inside the catch block.